internal/common/quota.go

```diff
@@ -0,0 +1,81 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/osbuild/image-builder/internal/db"
+)
+
+const (
+	day                  time.Duration = 24 * time.Hour
+	week                 time.Duration = 7 * day
+	DefaultSlidingWindow time.Duration = 2 * week
+	DefaultQuota         int           = 100
+)
+
+// the QUOTA_FILE needs to contain data arranged as such:
+//
+// {
+//     "000000":{
+//         "quota":2,
+//         "slidingWindow":1209600000000000
+//     },
+//     "000001":{
+//         "quota":0,
+//         "slidingWindow":1209600000000000
+//     },
+//     "default":{
+//         "quota":100,
+//         "slidingWindow":1209600000000000
+//     }
+// }
+//
+// The unit for the sliding window is the nanosecond.
+type Quota struct {
+	Quota         int           `json:"quota"`
+	SlidingWindow time.Duration `json:"slidingWindow"`
+}
+
+// Returns true if the number of requests made by accountNumber during a sliding window is below a threshold.
+// The duration of the sliding window and the value of the threshold must be set in a file pointed by the QUOTA_FILE
+// environment variable.
+func CheckQuota(accountNumber string, dB db.DB, quotaFile string) (bool, error) {
+	var authorizedRequests int
+	var slidingWindow time.Duration
+
+	// read proper values from quotas' file
+	var quotas map[string]Quota
+	jsonFile, err := os.Open(filepath.Clean(quotaFile))
+	if _, ok := err.(*os.PathError); ok {
+		fmt.Printf("No config file for quotas found at %s\n", quotaFile)
```

Comment:
We don't capture stdout/stderr, the logger object logs straight to cloudwatch. Either we return the patherror or `return false, fmt.Errorf(....)`